src/wallet/wallet.cpp

```diff
@@ -1456,6 +1488,39 @@ void CWallet::transactionRemovedFromMempool(const CTransactionRef& tx, MemPoolRe
         // https://github.com/bitcoin-core/bitcoin-devwiki/wiki/Wallet-Transaction-Conflict-Tracking
         SyncTransaction(tx, TxStateInactive{});
     }
+
+    const Txid& conflict_txid = tx->GetHash();
+
+    for (const CTxIn& tx_in : tx->vin) {
+        // If the wallet contains no transactions that spend the same prevout, continue
+        if (mapTxSpends.count(tx_in.prevout) == 0) continue;
+
+        std::pair<TxSpends::const_iterator, TxSpends::const_iterator> range;
+        range = mapTxSpends.equal_range(tx_in.prevout);
+
+        // For each wallet transaction spending this prevout...
+        for (TxSpends::const_iterator _it = range.first; _it != range.second; ++_it) {
+
+            auto try_updating_state = [&](CWalletTx& wtx) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet) {
+                if (!wtx.mempool_conflicts.empty()) {
+
+                    // Remove the previously conflicting transaction from this wtx's entry
+                    wtx.mempool_conflicts.erase(conflict_txid);
+
+                    if (wtx.mempool_conflicts.empty()) {
+                        // If this wtx has no other mempool conflicts, erase this wtx entry
```

Comment:
In 5fbf18b89d6768bed1be95dcbc98797df5788dfc "wallet: track mempool conflicts"

I'm a little bit confused by the "erase this wtx entry" part of the comment. AFAICT, no entries are being erased, Perhaps this should say:

```suggestion
                        // If this wtx has no other mempool conflicts, mark it as inactive
```

Reply:
Good point, this comment is from a previous version of the PR. I have fixed it.